iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1265,6 +1277,19 @@ static int setMqttMessagePropertyIfPossible(IOTHUB_MESSAGE_HANDLE IoTHubMessage,
     // Not finding a system property to map to isn't an error.
     int result = 0;
 
+    if (nameLen > 5)
```

Comment:
Is 5 b/c of the length of `ctime`?
In that case I would suggest 
```suggestion
    if (nameLen > strlen(MESSAGE_CREATION_TIME_UTC))
```
Since I don't know off the bat what `5` is here.

Reply:
Yes, it's an optimization of who ever wrote this function.
But there may be other strcmp in this if statement, so don't tie it to this one. See the rest of the implementation.